hudi-common/src/main/java/org/apache/hudi/exception/SchemaBackwardsCompatibilityException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.exception;
+
+import org.apache.hudi.avro.AvroSchemaCompatibility;
+
+import java.util.stream.Collectors;
+
+/**
+ * Thrown when there is a backwards compatibility issue with the incoming schema.
+ * i.e. when the incoming schema cannot be used to read older data files
+ */
+public class SchemaBackwardsCompatibilityException extends SchemaCompatibilityException {
```

Comment:
Seems no need to create a new one, or is this specific for Avro (renamed to `AvroIncompatibleSchemaException`)?

Reply:
We mostly follow the avro rules, but we have expanded them a bit. So I don't think it should include avro in the name

Reply:
In that case, name this as sth like `IncompatibleWriterSchemaOnReadException` to be more specific, to highlight that this is incompatibility between schema and data? Right now by looking at the class name it's hard to tell what this covers.  So either you can make the name more specific, or just use the general `SchemaCompatibilityException` with a util method to generate the exception instance.